jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/ResponseNotifier.java

```diff
@@ -368,6 +369,12 @@ public boolean isLast()
                     throw new UnsupportedOperationException();
                 }
 
+                @Override
+                public boolean canRetain()
+                {
+                    throw new UnsupportedOperationException();
```

Comment:
Hmmmm   EOF is also a sentinel instance, but we return true from it's isLast method rather than expecting callers to always do an instanceof check.

I guess if this chunk is never passed externally then OK.

Reply:
Yes it is never passed externally. It is only a marker to indicate that the chunk of one of the copies has been read, so further reads should return `null` until all copies have read the chunk.